frontend/src/app/settings/roles/roles.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, OnInit } from '@angular/core';
+import {ROLES, PERMISSIONS, IRoleTemp, IPermissionTemp} from './mock-roles';
+import {MatDialog} from '@angular/material/dialog';
+import { CreateRoleDialogComponent } from './create-role/create-role.component';
+import { UpdateRoleDialogComponent } from './update-role/update-role.component';
+import { DeleteEntryDialogComponent } from 'src/app/shared/delete-entry-dialog/delete-entry-dialog.component';
+
+interface LooseTableObject {
+  [key: string]: any;
+}
+
+
+@Component({
+  selector: 'app-roles',
+  templateUrl: './roles.component.html',
+  styleUrls: ['./roles.component.scss']
+})
+export class RolesComponent implements OnInit {
+
+  public tableData: LooseTableObject[] = [];
+  public displayedColumns: string[] = ['permission'];
+
+
+  constructor(public dialog: MatDialog) { }
+
+  public ngOnInit(): void {
+    // TODO:
+    // load data
+    this.createDynamicColums();
+    this.createTableData();
+
+  }
+
+  public openCreateDialog(): void {
+    const dialogRef = this.dialog.open(CreateRoleDialogComponent, {
+      data: PERMISSIONS
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        // TODO:
+        // call backend to add role
+        // call backend for all roles
```

Comment:
to do? when will it be done?

Reply:
https://github.com/SilvesterSchneider/CRohM/pull/72

wenn dieser pull request fertig ist -> then it will be done